packages/plugin-core/src/test/suite-integ/WorkspaceWatcher.test.ts

```diff
@@ -208,8 +215,55 @@ suite("WorkspaceWatcher: GIVEN the dendron extension is running", function () {
   });
 
   describe("GIVEN the user opening a file", () => {
+    let ext: IDendronExtension;
+
+    beforeEach(async () => {
+      ext = ExtensionProvider.getExtension();
+      await ext.activateWatchers();
+    });
+    afterEach(async () => {
+      // imporant since we activate workspace watchers
+      await ext.deactivate();
+    });
+    describeSingleWS(
+      "AND WHEN user opens non dendron file for the first time",
+      { ctx },
+      () => {
+        test("THEN do not affect frontmatter", async () => {
+          const { engine, vaults, wsRoot } = ExtensionProvider.getDWorkspace();
+          await FileTestUtils.createFiles(wsRoot, [{ path: "sample" }]);
+          const wsutils = new WSUtilsV2(ext);
+          const notePath = path.join(wsRoot, "sample");
+          const editor = await VSCodeUtils.openFileInEditor(
+            vscode.Uri.file(notePath)
+          );
+          const { onFirstOpen } =
+            UNSAFE_getWorkspaceWatcherPropsForTesting(ext);
+          expect(await onFirstOpen(editor)).toBeFalsy();
```

Comment:
minor nitpick, but this could also check that `foldFrontmatter` is not called since the main issue we're trying to fix is that Dendron tries to fold the frontmatter in non-note files. Or that could go into a different test if we want to keep the concerns of each test separate.

Reply:
yeah, that would be a better test. unfortunately not sure how to do this since in order to spy on a method, you need to either 1) pass in an object and the method or 2) spy on the method and inject the method

1) isn't feasible because the method is private so we can't spy on the object
2) isn't feasible because we don't have dependency inejction so we can't make `onFirstOpen` call a spied method

https://sinonjs.org/releases/latest/spies/